src/utils/cast.rs

```diff
@@ -0,0 +1,10 @@
+#[macro_export]
```

Comment:
this kind of macro appears useful to me. you commented that i should `ok_or` when deconstructing those variables, but unless i'm mistaken there isnt a nice inbuilt way to do that. given that we unwrap enums this way pretty frequently in the project, having a shorthand for it could save us some time. but open to suggestions either way